core/src/test/scala/org/apache/spark/ConsistentAccumulatorsSuite.scala

```diff
@@ -0,0 +1,284 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark
+
+import scala.ref.WeakReference
+
+import org.scalatest.Matchers
+
+import org.apache.spark.scheduler._
+
+
+class ConsistentAccumulatorSuite extends SparkFunSuite with Matchers with LocalSparkContext {
+  test("single partition") {
+    sc = new SparkContext("local[2]", "test")
+    val acc : Accumulator[Int] = sc.accumulator(0, consistent = true)
+
+    val a = sc.parallelize(1 to 20, 1)
+    val b = a.map{x => acc += x; x}
+    b.cache()
+    b.count()
+    acc.value should be (210)
+  }
+
+  test("adding only the first element per partition should work even if partition is empty") {
+    sc = new SparkContext("local[2]", "test")
+    val acc: Accumulator[Int] = sc.accumulator(0, consistent = true)
+    val a = sc.parallelize(1 to 20, 30)
+    val b = a.mapPartitions{itr =>
+      acc += 1
+      itr
+    }
+    b.count()
+    acc.value should be (30)
+  }
+
+  test("shuffled (combineByKey)") {
+    sc = new SparkContext("local[2]", "test")
+    val a = sc.parallelize(1 to 40, 5)
+    val buckets = 4
+    val b = a.map{x => ((x % buckets), x)}
+    val inputs = List(b, b.repartition(10), b.partitionBy(new HashPartitioner(5))).map(_.cache())
+    val mapSideCombines = List(true, false)
+    inputs.foreach{input =>
+      mapSideCombines.foreach{mapSideCombine =>
+        val accs = 1.to(4).map(x => sc.accumulator(0, consistent = true)).toList
+        val raccs = 1.to(4).map(x => sc.accumulator(0, consistent = false)).toList
```

Comment:
nit: ranges are an exception to the usual style rules, the preferred idiom is `(1 to 4)`.

Also, spaces around `{`, so `inputs.foreach { input => ...` etc.